v2/tools/generator/internal/config/configuration.go

```diff
@@ -312,6 +312,125 @@ func getModulePathFromModFile(modFilePath string) (string, error) {
+// For slices, appends new values to the end.
+// For maps, adds new key-value pairs but errors if attempting to overwrite existing keys.
+// For complex types, recursively applies the merge operation.
+// Returns an error if there are conflicts during merging.
+func (config *Configuration) Merge(other *Configuration) error {
+	if other == nil {
+		return nil
+	}
+
+	// Merge primitive fields
+	if err := mergePrimitiveString(&config.SchemaRoot, other.SchemaRoot, "SchemaRoot"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveString(&config.DestinationGoModuleFile, other.DestinationGoModuleFile, "DestinationGoModuleFile"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveString(&config.TypesOutputPath, other.TypesOutputPath, "TypesOutputPath"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveString(&config.TypeRegistrationOutputFile, other.TypeRegistrationOutputFile, "TypeRegistrationOutputFile"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveString(&config.RootURL, other.RootURL, "RootURL"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveString(&config.SamplesPath, other.SamplesPath, "SamplesPath"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveEnum(&config.Pipeline, other.Pipeline, "Pipeline"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveBool(&config.EmitDocFiles, other.EmitDocFiles, "EmitDocFiles"); err != nil {
+		return err
+	}
+
+	// Merge slice fields
+	config.AnyTypePackages = mergeSlice(config.AnyTypePackages, other.AnyTypePackages)
+	config.TypeFilters = mergeSlice(config.TypeFilters, other.TypeFilters)
+	config.TypeLoaderRenames = mergeSlice(config.TypeLoaderRenames, other.TypeLoaderRenames)
+	config.Transformers = mergeSlice(config.Transformers, other.Transformers)
+
+	// Merge complex fields
+	if err := config.Status.Merge(&other.Status); err != nil {
+		return err
+	}
+	
+	if other.SupportedResourcesReport != nil {
+		if config.SupportedResourcesReport == nil {
+			config.SupportedResourcesReport = NewSupportedResourcesReport(config)
+		}
+		if err := config.SupportedResourcesReport.Merge(other.SupportedResourcesReport); err != nil {
+			return err
+		}
+	}
+	
+	if other.ObjectModelConfiguration != nil {
+		if config.ObjectModelConfiguration == nil {
+			config.ObjectModelConfiguration = NewObjectModelConfiguration()
+		}
+		if err := config.ObjectModelConfiguration.Merge(other.ObjectModelConfiguration); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Helper functions for merging different types
+
+func mergePrimitiveString(base *string, other string, fieldName string) error {
+	if other == "" {
+		return nil // Nothing to merge
+	}
+	if *base == "" {
+		*base = other
+		return nil
+	}
+	if *base != other {
+		return fmt.Errorf("conflict in field %s: base value %q cannot be overwritten with %q", fieldName, *base, other)
+	}
+	return nil
+}
+
+func mergePrimitiveBool(base *bool, other bool, fieldName string) error {
+	if !other {
```

Comment:
This doesn't seem right?

If other sets false explicitly, base is true, we just don't merge?

That should be an error.

Reply:
I also think all of these `mergePrimitive...` could probably be written with generics in such a way that you didn't need 2 or 3 copies?

In fact the impl for `mergePrimitiveEnum` might work (though maybe the issue is `zero` handling?